addons/payment/models/payment_acquirer.py

```diff
@@ -618,6 +629,20 @@ def s2s_get_tx_status(self):
         return True
 
     @api.multi
+    def execute_callback(self):
+        res = None
+        for transaction in self.filtered(lambda tx: tx.callback_ir_model_id and tx.callback_res_id and tx.callback_method):
+            valid_token = transaction._generate_callback_hash()
+            if not consteq(ustr(valid_token), transaction.callback_hash):
+                _logger.warning('Receiving invalid callback for transaction %d' % (transaction.id))
```

Comment:
IMHO the message is misleading: it implies that the callback signature was transmitted through the acquirer and checked against the stored version - we only check that the signature generated at the time of the confirmation matches the one in the database.

"Invalid callback signature for transaction %d" % (transaction.id)

seems more truthful.